lib/livebook/notebook/explore/intro_to_nx.livemd

```diff
@@ -1,3 +1,731 @@
+implicitly, as before:
+
+```elixir
+TensorMath.subtract(tensor, Nx.tensor([1, 2]))
+```
+
+The broadcast worked as advertised, copying the `[1, 2]` row
+enough times to fill a `{2, 2}` tensor.  A tensor with a
+dimension of `1` will broadcast to fill the tensor:
+
+```elixir
+[[1], [2]] |> Nx.tensor() |> Nx.broadcast({1, 2, 2})
+```
+
+```elixir
+[[[1, 2, 3]]]
+|> Nx.tensor()
+|> Nx.broadcast({4, 2, 3})
+```
+
+Both of these examples copy parts of the tensor enough
+times to fill out the broadcast shape. You can check out the
+[Nx broadcasting documentation](https://github.com/elixir-nx/nx/blob/main/nx/lib/nx.ex#L47)
+for more details, or get `help` for the broadcast function:
+
+```elixir
+h(Nx.broadcast())
+```
+
+Much of the time, you won't have to broadcast yourself. Many of
+the functions and operators Nx supports will do so automatically.
+
+<!-- livebook:{"break_markdown":true} -->
+
+You now know how to build custom tensor-aware code.
+The next step is to make the code efficient on specialized CPUs
+using the backend of your choice.
+
+<!-- livebook:{"break_markdown":true} -->
+
+The last major feature we'll cover is called auto-differentiation, or autograd.
+
+## Automatic differentiation (autograd)
+
+An important mathematical property for a function is the
+rate of change, or the gradient. These gradients are critical
+for solving systems of equations and building probablistic
+models. In advanced math, derivatives, or differential equations,
+are used to take gradients. Nx can compute these derivatives
+automatically through a feature called automatic differentiation,
+or autograd.
+
+Here's how it works.
+
+<!-- livebook:{"disable_formatting":true} -->
+
+```elixir
+h Nx.Defn.grad
+```
+
+We'll build a module with a few functions,
+and then create another function to create the gradients of those
+functions. The function `grad/1` takes a function, and returns
+a function returning the gradient. We have two functions: `poly/1`
+is a simple numerical definitin, and `poly_slope_at/1` returns
+its gradient:
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+  poly: f(x) = 3x^2 + 2x + 1 \\
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+$$
+  polySlopeAt: g(x) = 6x + 2
+$$
+
+<!-- livebook:{"break_markdown":true} -->
+
+Here's the Elixir equivalent of those functions:
+
+```elixir
+defmodule Funs do
+  defn poly(x) do
+    3 * Nx.power(x, 2) + 2 * x + 1
+  end
+
+  def poly_slope_at(x) do
+    grad(&poly/1).(x)
+  end
+end
+```
+
+Notice the second `defn`. It uses `grad/1` to take its
+derivative using autograd. It uses the intermediate `defn` AST
+and mathematical composition to compute the derivative. You can
+see it at work here:
+
+```elixir
+Funs.poly_slope_at(2)
+```
+
+Nice. If you plug the number 2 into the function $$ 6x + 2 $$
+you get 14! Said another way, if you look at the graph at
+exactly 2, the rate of increase is 14 units of `poly(x)`
+for every unit of `x`, precisely at `x`.
```

Comment:
Great example! How hard do you think it would be for us to plot this using VegaLite? Perhaps we can have a chapter exclusively for autograd and we use vegalite for some plotting? 🤔

Reply:
I agree with the autograd and plots. Is it ok to get an initial commit on board, and track additional issues/improvements from there?